File: rest-service/src/main/java/mb/lib/queue/model/JobStatusResponse.java
package mb.lib.queue.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonSetter;
import mb.lib.model.JobStatus;

@JsonIgnoreProperties(ignoreUnknown = true)
public class JobStatusResponse
{
  private JobStatus status;

  @JsonSetter("status")
  public JobStatusResponse setStatus(JobStatus status) {
    this.status = status;
    return this;
  }

  @JsonIgnore
  public JobStatus getStatus() {
    return this.status;
  }
}

File: rest-service/src/main/kotlin/mb/lib/queue/model/JobStatusResponse.kt
package mb.lib.queue.model

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonSetter
import mb.lib.model.JobStatus

@JsonIgnoreProperties(ignoreUnknown = true)
data class JobStatusResponse(
  @get:JsonIgnore
  @set:JsonSetter("status")
  var status: JobStatus
)